### DISCUSSION TITLE
Maximum Frequency On Pin without Jitter On STM32F446

### DISCUSSION THREAD

**Discussion by compoundMR** [grblHAL/core#624] [Q&A] [answered]
i try to moves single pin on Axes Y and get frequency of 300 Khz Without Jitter , but when it comes into  moving 3 axes at the same time the frequency tend to decrease and when i see it , after 230Khz ( moving 3 axes ) the signal is jittering , did anyone ever try or have some explantion on why this happen ?
 
 thanks in advance

**Comment by terjeio** [comment, marked as answer]
The more axes step pulses has to be calculated and executed for the more time the processor has to spend doing so. This may affect interrupt latecy? Or even causing lost steps?
So there will always be an upper limit to max step frequency that is dependent on a number of factors and I believe it is impossible to determine that exacly without actually testing a range of different scenarioes. Other factors such as rate of motion commands received, size of planner buffer and other stuff that increases the load.

Ideally a second MCU, or a FPGA, should be employed for step generation in order to make it more deterministic...
